lightning/src/ln/peer_handler.rs

```diff
@@ -360,8 +390,11 @@ pub struct PeerManager<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, L: De
 	logger: L,
 }
 
-enum MessageHandlingError {
+/// An error indicating a failure to handle a received message.
+pub enum MessageHandlingError {
```

Comment:
Actually, on second thought, can we not expose this and just have `CustomMessageHandler`s return `LightningError`s? `LightningError` should cover all possible cases that users will want to trigger, `PeerHandleError` is somewhat more of a thing that we return to users from `PeerManager`, making both be used for receiving errors from users and returning errors to use blurs them a bit much IMO.

Reply:
Works for me, done.